controlplane/kubeadm/internal/controllers/upgrade.go

```diff
@@ -74,6 +74,16 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, errors.Wrap(err, "failed to update the kubernetes version in the kubeadm config map")
 	}
 
+	// FIXME(chrischdi): use a better semver comparison
+	// FIXME(chrischdi): think about having a better if structure 😅
+	if parsedVersion.Minor == 25 &&
+		(kcp.Spec.KubeadmConfigSpec.ClusterConfiguration == nil || kcp.Spec.KubeadmConfigSpec.ClusterConfiguration.ImageRepository == "") {
+		// FIXME(chrischdi): use a const instead of hardcoding
+		if err := workloadCluster.UpdateImageRepositoryInKubeadmConfigMap(ctx, "registry.k8s.io", parsedVersion); err != nil {
+			return ctrl.Result{}, errors.Wrapf(err, "failed to update the image repository in the kubeadm config map to %s for v1.25 compatibility", "registry.k8s.io")
+		}
+	}
+
 	if kcp.Spec.KubeadmConfigSpec.ClusterConfiguration != nil {
```

Comment:
Also, wondering if we can combine above change with the current one, so we update the config map only once